bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/ProcessASS.java

```diff
@@ -0,0 +1,219 @@
+package org.dataflowanalysis.analysis.converter;
+
+import java.util.HashMap;
+import java.util.List;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+import org.dataflowanalysis.analysis.core.*;
+import org.dataflowanalysis.analysis.pcm.core.AbstractPCMActionSequenceElement;
+
+import org.dataflowanalysis.analysis.pcm.core.user.*;
+import org.palladiosimulator.pcm.core.entity.Entity;
+
+import org.dataflowanalysis.dfd.datadictionary.*;
+import org.dataflowanalysis.dfd.dataflowdiagram.*;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.eclipse.emf.ecore.xmi.impl.XMIResourceFactoryImpl;
+
+@SuppressWarnings({"unchecked","rawtypes"})
+public class ProcessASS {
+	
+	private Map<Entity, Node> dfdNodeMap = new HashMap<>();
+	private DataDictionary dd = datadictionaryFactory.eINSTANCE.createDataDictionary();
+	private DataFlowDiagram dfd = dataflowdiagramFactory.eINSTANCE.createDataFlowDiagram();
+			
+	public DataDictionary getDictionary() {
+		return this.dd;
+	}
+	
+	public DataFlowDiagram getDataFlowDiagram() {
+		return this.dfd;
+	}
+	
+	public void transform(List<ActionSequence> ass) {
+		for (ActionSequence actionSequence : ass) {
+			Node previousNode = null;
+			for (AbstractActionSequenceElement<?> ASE : actionSequence.getElements()) {
+				if(ASE instanceof AbstractPCMActionSequenceElement) {
+					previousNode = processActionSequenceElement((AbstractPCMActionSequenceElement)ASE, previousNode);
+				}
+			}
+		}
+	}
+	
+	private Node processActionSequenceElement(AbstractPCMActionSequenceElement pcmASE, Node previousDFDNode) {
+		Node dfdNode = getOrCreateDFDNode(pcmASE);
+		
+		// Add a flow between previous node and current node
+		createFlows(previousDFDNode, dfdNode, pcmASE);
+		
+		return dfdNode;
+	}
+	
+	
+	private void createFlows(Node source, Node dest, AbstractPCMActionSequenceElement pcmASE) {
+		if(source == null || dest == null) {
+			return;
+		}
+		List<DataFlowVariable> FlowVariables = pcmASE.getAllDataFlowVariables();
+		for(DataFlowVariable flowVariable : FlowVariables) {
+			String flowName = flowVariable.variableName();
+			
+			Optional<Flow> optFlow = dfd.getFlows().stream().filter(f -> (
+					f.getSourceNode().equals(source) && 
+					f.getDestinationNode().equals(dest) &&
+					f.getEntityName().equals(flowName))).findFirst();
```

Comment:
Replace this the filter expression with multiple filter expressions to improve readability

Reply:
Likes this?
```
Optional<Flow> optFlow = dfd.getFlows().stream()
    .filter(f -> f.getSourceNode().equals(source))
    .filter(f -> f.getDestinationNode().equals(dest))
    .filter(f -> f.getEntityName().equals(flowName))
    .findFirst();
```